src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -171,59 +170,153 @@ class IpexCommunicationService extends AgentService {
   }
 
   @OnlineOnly
-  async offerAcdcFromApply(notification: KeriaNotification, acdc: any) {
-    const msgSaid = notification.a.d as string;
-    const msg = await this.props.signifyClient.exchanges().get(msgSaid);
+  async offerAcdcFromApply(id: string, acdc?: any) {
+    const applyNoteRecord = await this.notificationStorage.findById(id);
+
+    if (!applyNoteRecord) {
+      throw new Error(
+        `${IpexCommunicationService.NOTIFICATION_NOT_FOUND} ${id}`
+      );
+    }
+
+    if (Object.keys(applyNoteRecord.linkedGroupRequests).length) {
```

Comment:
This is a little different to when we were accepting an ACDC. If we have 2 possible different credentials we can offer to an apply, FIRST and SECOND wallet can choose the different credentials and THIRD can join either one of them. The user has to choose who to join.

So for offer, I think we need to do this:
```
linkedGroupRequests: {
  credentialSaid1: {
    accepted: true,
    saids: [offerSaid1, offerSaid2]
  },
  credentialSaid2: {
    accepted: false,
    saids: [offerSaid3, offerSaid4]
  }
}
```

Reply:
So we go a layer deeper.